int32 is_arrived
int32 mode1_finish